impl/src/main/java/jakarta/faces/component/html/HtmlEvents.java

```diff
@@ -0,0 +1,185 @@
+package jakarta.faces.component.html;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import jakarta.faces.component.ActionSource;
+import jakarta.faces.component.EditableValueHolder;
+import jakarta.faces.component.behavior.ClientBehaviorHolder;
+import jakarta.faces.context.FacesContext;
+import jakarta.faces.event.BehaviorEvent.FacesComponentEvent;
+
+/**
+ * <p class="changed_added_5_0">
+ * Events supported by HTML elements as per <a href="https://html.spec.whatwg.org/multipage/webappapis.html#event-handlers-on-elements,-document-objects,-and-window-objects">current spec</a>.
+ * These can be used to supply {@link ClientBehaviorHolder#getEventNames()} and {@link ClientBehaviorHolder#getDefaultEventName()}.
+ * </p>
+ *
+ * @since 5.0
+ */
+public final class HtmlEvents {
+
+    /**
+     * Events supported by all HTML document elements.
+     */
+    public enum HtmlDocumentElementEvent {
+        abort,
+        auxclick,
+        beforeinput,
+        beforematch,
+        eforetoggle,
```

Comment:
Spelling mistake should be `beforeToggle` ?

Reply:
Eew, well spotted, fixed now and thanks!